### DISCUSSION TITLE
Creating specific menu items and Collection menu items.

### DISCUSSION THREAD

**Discussion by weishuhn** [windingwind/zotero-plugin-toolkit#19] [Q&A] [answered]
I am writing a plugin to integrate my website, [Inciteful](https://inciteful.xyz) into Zotero with a plugin: https://github.com/inciteful-xyz/inciteful-zotero-plugin

I have two questions...

1) I was wondering if there was a way to create a right click menu item for only top level items rather than for everything like notes and attachments as well.  

2) Is there a way to create a menu item targeting a collection right click menu in the "My Library" pane?

Thanks!

**Comment by weishuhn** [comment, marked as answer]
Never mind I figured it out.  

For question 1 I just used the `Item.topLevelItem` property in order to make sure I was getting the information I needed. 

For question 2, there was a `collection` option that I had missed which I passed into the register method like so:
``` js
ztoolkit.Menu.register('collection', {
      tag: 'menuitem',
      id: 'zotero-collectionmenu-inciteful-search',
      label: getString('menuitem.search'),
      commandListener: ev => addon.hooks.onSearchCollectionEvent(ev),
      icon: menuIcon
})
```

Hope this helps someone!

**Reply by quinn-p-mchugh** [reply]
@weishuhn 
This helped me immensely. Thank you for posting your solution!
